### ISSUE TITLE
[mesheryctl] Install: Support new Meshery Adapter for Nighthawk

### ISSUE THREAD

**Issue by leecalcote** [meshery/meshery#10371]
_See [mesheryctl Command Tracker](https://bit.ly/3dqXy1q) for current status of commands._

### Current Behavior
A new Meshery Adapter is being created in the v0.8 release: https://github.com/meshery/meshery-nighthawk

### Desired Behavior
`mesheryctl system` commands (start, stop, restart, logs and context) all need to support this new adapter.

### Acceptance Tests
1. `mesheryctl system context create --components` supports this new adapter.
2. `mesheryctl system start`
3. `mesheryctl system stop`
4. `mesheryctl system ....` and others

---

### Contributor Resources

- ⌨️ [mesheryctl Contributing Guide](https://github.com/meshery/meshery/blob/master/mesheryctl/README.md)
- [Beginner's guide to contributing to Meshery and mesheryctl](https://youtu.be/hh_kFLZx3G4)
- [mesheryctl Command Tracker](https://docs.google.com/spreadsheets/d/1q63sIGAuCnIeDs8PeM-0BAkNj8BBgPUXhLbe1Y-318o/edit#gid=0)
- [Meshery CLI Commands and Documentation](https://docs.google.com/document/d/1xRlFpElRmybJ3WacgPKXgCSiQ2poJl3iCCV1dAalf0k/edit#heading=h.5fucij4hc5wt)
- 🙋🏾🙋🏼 Questions: [Layer5 Discussion Forum](http://discuss.meshery.io) and [Layer5 Community Slack](https://slack.meshery.io)

**Comment by MeenuyD** [comment]
Hello @leecalcote I would like to work on this issue can you please assign it to me.

**Comment by lakshz** [comment]
I would to work on this. Can this be assigned to me?

**Comment by sandramsc** [comment]
@lakshz Any update/progress on the issue?

**Comment by KunyueXing** [comment]
Hi @leecalcote, I'd like to work on this issue. Could you please assign this to me? I've created a PR for Sistent issue [#5486](https://github.com/meshery/meshery/issues/5486), so now I've got time to take a new one.

I noticed that this issue has been opened for 3 months with no updates. Is this task still needed?

**Comment by leecalcote** [comment]
@gyohuangxin what do you think? Will we be able to press forward here?

**Comment by gyohuangxin** [comment]
@leecalcote Yes, as https://github.com/meshery/meshery-nighthawk/releases/tag/v0.7.1 released, we need to enable meshery-nighthawk installation in mesheryctl.
@KunyueXing Thanks for taking this issue.

**Comment by KunyueXing** [comment]
@sandramsc @leecalcote @gyohuangxin  Thanks for assigning this issue to me. I'm starting working on this and will keep you updated.

**Comment by KunyueXing** [comment]
Here's some homework that I did in the past week:

1. Watched [beginners' guide to contributing to Meshery and MesheryCtl](https://www.youtube.com/watch?v=hh_kFLZx3G4)
2. [Meshery contribution guide](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md) and [MesheryCtl contribution guide](https://github.com/meshery/meshery/blob/master/mesheryctl/README.md) 
3. Installed necessary languages, tools, and set up the environment.
4. Getting familiar with Golang, especially the coding style guidelines ([code review comments](https://go.dev/wiki/CodeReviewComments), and [go wiki](https://go.dev/wiki/)) and [the best practices for formatting and style](https://peter.bourgon.org/go-in-production/#formatting-and-style).
5. Getting familiar with [Cobra](https://github.com/spf13/cobra) framework, and [Cobra concepts](https://github.com/spf13/cobra#concepts).
6. Getting familiar with the code base of [`mesheryctl system` commands](https://github.com/meshery/meshery/tree/master/mesheryctl/internal/cli/root/system).

**Comment by gyohuangxin** [comment]
@KunyueXing Excellent job!

**Comment by KunyueXing** [comment]
Here's some updates,

- Although in the description, it mentioned "Adapter", I found that in `mesheryctl` code, all of the mentioning of adapter have been changed to `component`. It took me a while to locate the code that deals with adapters/components
- to support this adapter in `mesheryctl system start`,
  - so far, I found 2 places that need changing, 
    - for `docker` platform, the Nighthawk adapter needs to be added as a new `Service` to `utils.Services` at [mesheryctl/pkg/utils/helpers.go:280](https://github.com/meshery/meshery/blob/master/mesheryctl/pkg/utils/helpers.go#L280)
    - for `kubernetes` platform, the Nighthawk helm sub-chart needs to be included in [SetOverrideValues()](https://github.com/meshery/meshery/blob/master/mesheryctl/pkg/utils/helpers.go#L1108)
  - there're probably more places to change and I'll continue my investigation

Here's my question,

1. to add the Nighthawk adapter as a new `Service`, I need its `Image`, `Labels`, and `Ports`. Its `Labels` seems to be `[]string{"com.centurylinklabs.watchtower.enable=true"}`, and `Ports` seems to be `[]string{"10013:10013"}`. But I wasn't able to find its `Image`. @leecalcote and @gyohuangxin, could you please let me know how to find the image? There's seemingly a relevant page ([meshery/getnighthawk](https://hub.docker.com/r/meshery/getnighthawk)), but it's empty.

**Comment by gyohuangxin** [comment]
@KunyueXing Thank you for your findings regarding the support of the Nighthawk adapter in `meshery system start`, and please go ahead to create issues, PRs to implement your findings. I think the more findings will be emerge during your implementation. 😊

> to add the Nighthawk adapter as a new Service, I need its Image, Labels, and Ports. Its Labels seems to be []string{"com.centurylinklabs.watchtower.enable=true"}, and Ports seems to be []string{"10013:10013"}. But I wasn't able to find its Image. @leecalcote and @gyohuangxin, could you please let me know how to find the image? There's seemingly a relevant page ([meshery/getnighthawk](https://hub.docker.com/r/meshery/getnighthawk)), but it's empty.

Here is the place the meshery-nighthawk images should be located: https://hub.docker.com/r/meshery/meshery-nighthawk/tags . There is a old image for the test, if you have any issues with it please contact us to upload a new one.

**Comment by KunyueXing** [comment]
Hi @gyohuangxin, I appreciate your encouraging and helpful reply in this issue. Thank you very much for all the information and help. :)

**Comment by userAdityaa** [comment]
Hey @KunyueXing, is this still in progress ?

**Comment by naveensrinivas282** [comment]
Hey, @leecalcote does this issue still need work ? I'd like to pick it up if it's still pending

**Comment by leecalcote** [comment]
@naveen28204280, yes, it does. Produce a design spec and add to an upcoming Meshery dev meeting agenda. We'll get the issue assigned thereafter.

**Comment by Junnygram** [comment]
Hello @leecalcote and @gyohuangxin ,

I would love to help out and implement this issue. I have put together the Functional Specification covering the `mesheryctl` integration logic here: 
[Functional Specification: Support Meshery Adapter for Nighthawk](https://docs.google.com/document/d/1Y_elDk4RE3Q8z6_-jIjane8s6mbqwlTZKTGGSvnXPn0/edit?usp=sharing)

I will also add it to the agenda for the upcoming Meshery development meeting. Looking forward to discussing it!

**Comment by shafiankhan** [comment]
Hi @leecalcote @gyohuangxin,

I’ve picked up this issue and submitted a PR implementing the Nighthawk adapter support:

PR: #21946

I reviewed the existing mesheryctl adapter/component registration flow and implemented the required support for `meshery-nighthawk`.

### Changes included
- Registered `meshery-nighthawk` in `mesheryctl/pkg/utils/helpers.go`
- Added the Nighthawk adapter URL on port `10013`
- Added the adapter to `SetOverrideValues()`
- Updated `TestSetOverrideValues` coverage
- Added the Nighthawk adapter configuration to the Meshery Helm values

The PR is intentionally limited to these changes. I did not add a Helm dependency because a Nighthawk Helm sub-chart is not currently present in the repository.

I also reviewed the existing Functional Specification linked in this issue and used it as the basis for the implementation.

I’d appreciate a review of the design and implementation, especially around:
- Whether the existing Functional Specification is sufficient for this implementation
- The Nighthawk image/tag being used
- Whether the Helm configuration should remain in this PR or be handled separately

PR: #21946

Thanks!
